src/ape/exceptions.py

```diff
@@ -0,0 +1,309 @@
+from typing import List, Set, Type
+
+
+class ApeError(Exception):
```

Comment:
Prefer `ApeException` for the base class case

Reply:
Also wonder if we should subclass `Abort` too from `click` :thinking:, so all unhandled exceptions cause a CLI Abort by default

Reply:
Yes, we could do that way.
It does have that behavior currently because of the overridden `invoke()` method that catches it and raises `Abort`.
, but perhaps using that base class is easier.

I was trying to not worry about click or CLI related stuff for the exceptions module though, but maybe unnecessarily

Reply:
Yeah, I mean maybe not super over-complicate it, but if there's a way it makes sense to simplify our handling of the command line, this would be the way to do it

Reply:
We could subclass `Abort` (or just move `Abort` to `exceptions` and rename it to `ApeException`) but we can not use the shared util method `notify` anymore.... Is that better or worse than the way it is now, where we catch `ApeException` and raise `Abort`? I could go either way.

Reply:
for sure, I guess we can add a blanket `except` for `ApeException` and `raise Abort(...) from e` for that